packages/babel-core/test/option-manager.js

```diff
@@ -11,6 +11,113 @@ describe("option-manager", () => {
     }, /Babel 5 plugin is being run with an unsupported Babel/);
   });
 
+  describe("config plugin/preset flattening and overriding", () => {
+    function makePlugin() {
+      const calls = [];
+      const plugin = (api, opts) => {
+        calls.push(opts);
+        return {};
+      };
+      return { plugin, calls };
+    }
+
+    it("should throw if a plugin is repeated", () => {
+      const { calls, plugin } = makePlugin();
+
+      assert.throws(() => {
+        manageOptions({
+          plugins: [plugin, plugin],
+        });
+      }, /Duplicate plugin\/preset detected/);
+      assert.deepEqual(calls, []);
+    });
+
+    it("should not throw if a repeated plugin has a different name", () => {
+      const { calls: calls1, plugin: plugin1 } = makePlugin();
+      const { calls: calls2, plugin: plugin2 } = makePlugin();
+
+      manageOptions({
```

Comment:
isnt this a private API? maybe we could try to express the test using public API exclusively

Reply:
Yeah I can rename this. This is the same public API you can get from `babel.loadOptions` so I think I can change these tests to use that.